### ISSUE TITLE
TApplication::onInitComplete

### ISSUE THREAD

**Issue by javalizard** [pradosoft/prado#488]
We propose a new event in TApplication called onInitComplete.   onInitComplete is best called in initApplication before starting to process any services.  In this way, services may be added dynamically by any modules.  This is the method for doing any post initialization clean up.

This method is fundamental to application initialization of prado-cli commands.  onBeginRequest is NOT called during CLI for these requests.  onBeginRequest has been the event that developers are told to bind upon for any module post-initialization.  During CLI initialization, all modules requiring onBeginRequest are not properly initialized.

Of course, PRADO documentation (pictures and text) would need to be updated.

Like this:
TApplication.php

```
protected function initApplication()
{
    Prado::trace('Initializing application','System.TApplication');

    if($this->getConfigurationFile()!==null)
    {
        $_cacheFile=$this->getRuntimePath().DIRECTORY_SEPARATOR.TApplication::CONFIGCACHE_FILE;
        if($_cacheFile===null || @filemtime($_cacheFile)<filemtime($this->getConfigurationFile()))
        {
            $config=new TApplicationConfiguration;
            $config->loadFromFile($this->getConfigurationFile());
            if($_cacheFile!==null)
                file_put_contents($_cacheFile,Prado::serialize($config),LOCK_EX);
        }
        else
            $config=Prado::unserialize(file_get_contents($_cacheFile));

        $this->applyConfiguration($config,false);
    }

    $this->onInitComplete();

    if(($serviceID=$this->getRequest()->resolveRequest(array_keys($this->_services)))===null)
        $serviceID=$this->getPageServiceID();

    $this->startService($serviceID);
}

/**
 * Raises onInitComplete event.
 * At the time when this method is invoked, application modules are loaded, 
 * and the application is about to start processing the user request.
 */
public function onInitComplete()
{
    $this->raiseEvent('onInitComplete',$this,null);
}
```

**Comment by belisoful** [comment]
Submitted a Pull Request for resolution of this enhancement.  I would like to call this a bug because Shell applications have no place for post initialize completion except for this.

the graphic on http://www.pradoframework.net/demos/quickstart/?page=Fundamentals.Applications needs to be updated too.

**Comment by ctrlaltca** [comment]
fixed in #751
